src/core/compact_object.h

```diff
@@ -410,13 +426,22 @@ class CompactObj {
     mask_ = mask;
   }
 
+  // Must be 16 bytes.
   struct ExternalPtr {
-    uint32_t type : 8;
-    uint32_t reserved : 24;
-    uint32_t page_index;
+    uint32_t serialized_size;
     uint16_t page_offset;  // 0 for multi-page blobs. != 0 for small blobs.
-    uint16_t reserved2;
-    uint32_t size;
+    uint16_t is_cool : 1;
+    uint16_t is_reserved : 15;
+
+    struct Offload {
+      uint32_t page_index;
+      uint32_t reserved;
+    };
+
+    union {
+      Offload offload;
+      detail::TieredColdRecord* cold_record;
+    };
```

Comment:
Are we forced to store the page_index inside cold_record because otherwise we don't fit into 16 bytes? It's somewhat confusing this way

Reply:
yes, to reduce the memory requirements

Reply:
Let's comment it somewhere, becuase I trying to find other logic begind it 🙂